### PR TITLE
feat: enable resource identifier composing from multiple variables

### PR DESCRIPTION
**Pull request by FemiNoviaLina**
Enable resource identifier to be constructed from multiple variable using templating.

### CODE REVIEW

File: internal/proxy/helper.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package proxy
```
Comment (mabdh) [inline internal/proxy/helper.go]: What about creating a new package called `attribute` inside proxy and do attribute manipulation in it?

File: internal/proxy/attribute/attribute.go [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
+	"github.com/valyala/fasttemplate"
+)
 
 const (
-	AttributeTypeQuery       AttributeType = "query"
-	AttributeTypeHeader      AttributeType = "header"
 	AttributeTypeJSONPayload AttributeType = "json_payload"
 	AttributeTypeGRPCPayload AttributeType = "grpc_payload"
-	AttributeTypePathParam   AttributeType = "path_param"
+	AttributeTypeQuery       AttributeType = "query"
```
Comment (mabdh) [inline internal/proxy/attribute/attribute.go]: We can refactor into something like this
```suggestion
    TypeQuery      Type = "query"
```

File: internal/proxy/attribute/attribute.go [hunk 3]
```diff
@@ -18,4 +27,13 @@ type Attribute struct {
 	Path   string        `yaml:"path" mapstructure:"path"`
 	Params []string      `yaml:"params" mapstructure:"params"`
 	Value  string        `yaml:"value" mapstructure:"value"`
+	Source string        `yaml:"source" mapstructure:"source"`
+}
+
+func ComposeAttribute(attribute string, attrs map[string]interface{}) string {
+	if strings.Contains(attribute, "{") {
+		template := fasttemplate.New(attribute, "{", "}")
```
Comment (mabdh) [inline internal/proxy/attribute/attribute.go]: Should use double curly bracket for the delimiter

```suggestion
		template := fasttemplate.New(attribute, "{{", "}}")
```

File: internal/proxy/hook/authz/authz.go [hunk 4]
```diff
@@ -18,6 +18,7 @@ import (
 	"github.com/goto/shield/core/relation"
 	"github.com/goto/shield/core/resource"
 	"github.com/goto/shield/core/user"
+	proxyattr "github.com/goto/shield/internal/proxy/attribute"
```
Comment (mabdh) [inline internal/proxy/hook/authz/authz.go:21]: Why should we use`proxyattr`?

Comment (FemiNoviaLina) [reply internal/proxy/hook/authz/authz.go:21]: in this file, attribute package from opentelemetry is imported so we use `proxyattr` alias.

File: internal/proxy/attribute/attribute.go (hunk 3 above)
Comment (FemiNoviaLina) [reply internal/proxy/attribute/attribute.go]: use ${} instead so yaml parser won't assume the value as map

File: internal/proxy/attribute/attribute.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	Key    string        `yaml:"key" mapstructure:"key"`
+	Type   AttributeType `yaml:"type" mapstructure:"type"`
+	Index  string        `yaml:"index" mapstructure:"index"` // proto index
+	Path   string        `yaml:"path" mapstructure:"path"`
+	Params []string      `yaml:"params" mapstructure:"params"`
+	Source string        `yaml:"source" mapstructure:"source"`
+	Value  string        `yaml:"value" mapstructure:"value"`
+}
+
+func ComposeAttribute(attribute string, attrs map[string]interface{}) string {
```
Comment (mabdh) [inline internal/proxy/attribute/attribute.go]:
```suggestion
func Compose(attribute string, attrs map[string]interface{}) string {
```

**Review by mabdh — APPROVED** [review]

**Pull request merged**